### DISCUSSION TITLE
Run-ALPipeline script override examples?

### DISCUSSION THREAD

**Discussion by cegekaJG** [microsoft/AL-Go#1078] [Q&A] [answered]
I'm interested in experimenting with overriding parts of the AL Pipeline, but the documentation doesn't explain much about how to effectively implement a custom script. Is there a way to review the `params` hashtable at this point? Do I have to return a set of specific variables for the pipeline to run smoothly? Are there any examples of different overrides?

**Comment by KristofKlein** [comment]
I looked a bit around in the BusinessCentral core app repo. there they do an overwrite of the NewBCContainer.ps1. I would guess this is some sort of what it has to look like to run like no overwrite.

Param(
    [Hashtable]$parameters
)

# Add Custom Stuff here 

New-BcContainer @parameters

# Or here 

Invoke-ScriptInBcContainer -containerName $parameters.ContainerName -scriptblock { $progressPreference = 'SilentlyContinue' }

now you can add things right in front of the New-BcContainer to influence things 

$parameters.multitenant = $true
$parameters.RunSandboxAsOnPrem = $true
$parameters.memoryLimit = "16G"
if ("$env:GITHUB_RUN_ID" -eq "") {
    $parameters.isolation = "hyperv"
    $parameters.includeAL = $true
    $parameters.doNotExportObjectsToText = $true
    $parameters.shortcuts = "none"
}

Just place it in the .AL-Go folder ....based on the doc you can do that for several other files as well... not sure it helps with the Run-ALPipeline script ... :)

**Comment by cegekaJG** [comment, marked as answer]
I completely forgot about this thread; I've since created my own custom script to deploy to OnPrem environments.
You can check it out here: https://github.com/cegekaJG/AL-Go/blob/cegeka/Templates/Per%20Tenant%20Extension/.github/DeployToOnPrem.ps1

**Comment by freddydk** [comment]
You will also see some script overrides used in the BCApps repository, here:

https://github.com/microsoft/BCApps/tree/main/build/projects/System%20Application%20Tests/.AL-Go

(or in the other projects in that repo)
